.github/workflows/tag-version.yml

```diff
@@ -0,0 +1,102 @@
+        required: false
+        default: '@("${{ github.repository }}")'
+        description: >
+          PowerShell string array of repository base URLs to include when evaluating called GHA Workflows and Actions,
+          e.g '@("Lombiq/GitHub-Actions")' or '@("Lombiq/GitHub-Actions",
+          "Lombiq/Open-Source-Orchard-Core-Extensions")'. The parameter must be a PowerShell string array.
+        type: string
+      additional-pattern-include-list:
+        required: false
+        default: '@()'
+        description: >
+          PowerShell string array of additional RegEx patterns to include when searching for branch references that need
+          to be updated, e.g.
+          'https://raw.githubusercontent.com/Lombiq/GitHub-Actions/(?<ref>[aA-zZ,0-9,-,\.,/,_]*)/.github/'. The pattern
+          MUST include a regex named capture group (?<ref>[aA-zZ,0-9,-,\.,/,_]*) so the captured ref can be updated
+          correctly. The parameter must be a PowerShell string array.
+        type: string
+      expected-ref:
+        required: false
+        default: '${{ github.ref_name }}'
+        description: The expected reference value to set for all called GHA Workflows and Actions.
+        type: string
+
+jobs:
+  run:
+    name: Set GitHub Action/Workflow to Version Tag
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        shell: pwsh
+    steps:
+      - name: Setup Scripts
+        run: |
+          "${{ github.action_path }}" >> $Env:GITHUB_PATH
+          (Resolve-Path "${{ github.action_path }}/../../../Scripts").Path >> $Env:GITHUB_PATH
+
+      - name: Checkout Repository
+        uses: Lombiq/GitHub-Actions/.github/actions/checkout@dev
+        with:
+          token: ${{ secrets.TAG_VERSION_TOKEN }}
+
+      - name: Determine Version Tag Name from Branch Name
+        id: determine-tag
+        run: |
+          $refname = "${{ inputs.expected-ref }}"
+          $tagname = $refname -replace 'release/', ''
+          Set-GitHubOutput -Key 'output' -Value $tagname
+
+      - name: Set Ref for GitHub Actions and Workflows
+        uses: Lombiq/GitHub-Actions/.github/actions/set-gha-refs@issue/OSOE-735
+        with:
+          path-include-list: ${{ inputs.path-include-list }}
+          file-include-list: ${{ inputs.file-include-list }}
+          called-repo-base-include-list: ${{ inputs.called-repo-base-include-list }}
+          additional-pattern-include-list: ${{ inputs.additional-pattern-include-list }}
+          expected-ref: '${{ steps.determine-tag.outputs.tagname }}'
+
+      - name: Commit and Push Changes
+        uses: stefanzweifel/git-auto-commit-action@8756aa072ef5b4a080af5dc8fef36c5d586e521d # v5
+        with:
+          commit_message: 'Set GitHub Actions/Workflows to tag ${{ steps.determine-tag.outputs.tagname }}'
+          tagging_message: ${{ steps.determine-tag.outputs.tagname }}
+
+      - name: Force Push Tag
+        continue-on-error: true
+        run: git push --tags --force
```

Comment:
Also publish a release on GitHub, please. We do this [during NuGet publishing too](https://github.com/Lombiq/GitHub-Actions/blob/dev/.github/actions/publish-nuget/action.yml#L216). Basically this should suffice:

```yaml
    - name: Create Release
      # v1.11.2
      uses: ncipollo/release-action@18eadf9c9b0f226f47f164f5373c6a44f0aae169
      # This is to prevent creating releases when pushing tags for issue-specific pre-releases like
      # v4.3.1-alpha.osoe-86.
      if: "!contains(steps.setup.outputs.publish-version, '-')"
      with:
        allowUpdates: true
        generateReleaseNotes: true
```

However, at this point, it would be worth factoring it out to a central internally used action, so it's easier to keep its version updated all at once (instead of these two places), much like how I did it here: https://github.com/Lombiq/GitHub-Actions/pull/336/files.

Reply:
Ok, I can add that in and look to centralize that into it's own internal action.

Reply:
Thanks!

Reply:
Refactored to internal action `release-action` in e0ce7b52744b17f81bbd90f7c60ea9b9afc43547.